### ISSUE TITLE
Implementation of filter function

### ISSUE THREAD

**Issue by johartl** [PokemonGoers/PokeMap-1#21]
Your map needs to implement the filter function. Specification from https://github.com/PokemonGoers/Catch-em-all/issues/24
- `filter({ pokemonIds, sightingsSince, predictionsUntil })` applies filter options to the map.
  - `pokemonIds` Array of Pokemon IDs which are shown (applies to predictions and sightings) on the map. If `null` show all 151 Pokemon.
  - `sightingsSince` Time in seconds for which sightings (includes mobs) in the past are shown. If `0` don't show any sightings.
  - `predictionsUntil` Time in seconds for which predictions are shown. If `0` don't show any predictions.

The filter function will make your `updateTimeRange` function obsolete.

**Comment by gqinami** [comment]
@johartl just for clarifications, would there be cases where we show predictions and sightings at the same time, or it is just one or the other??

Thanks

**Comment by johartl** [comment]
``` (json)
{
  sightingsSince: 3600,
  predictionsUntil: 1800,
  pokemonIds: null
}
```

In this case you would show sightings for the last hour and predictions for the next 30 minutes. So yes, there can be both :)

**Comment by johartl** [comment]
Are you guys still working on this?

**Comment by gqinami** [comment]
@johartl yes we are working on it. Will let you know soon when its done

**Comment by johartl** [comment]
Is this done guys?

**Comment by gqinami** [comment]
Yes it is done. Only predictions are not there yet, since we havent got the data for it.

**Comment by johartl** [comment]
Okay that's great! So I will keep this issue open until the predictions filtering is implemented.
